src/posix/platform/openthread-core-posix-config.h

```diff
@@ -287,4 +287,16 @@
 #define OPENTHREAD_CONFIG_LOG_PREPEND_UPTIME 1
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_MAX_SERVICES
+ *
+ * Specifies number of service entries in the SRP client service pool.
+ *
+ * This config is applicable only when `OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_ENABLE` is enabled.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_MAX_SERVICES
+#define OPENTHREAD_CONFIG_SRP_CLIENT_BUFFERS_MAX_SERVICES 20
```

Comment:
It goes from 10 to 20. Each requiring about ~300 bytes (so 3000 bytes). 
Posix platform in general is not expected to be resource limited.
 
We need this to be able to run the newly added test under `posix` build (otherwise we need to explicitly skip this test under `posix` (which I prefer not to).